app/models/vm_or_template.rb

```diff
@@ -1752,8 +1752,17 @@ def self.tenant_id_clause(user_or_group)
     vm_tenant_ids       = Vm.accessible_tenant_ids(user_or_group, Rbac.accessible_tenant_ids_strategy(Vm))
     return if template_tenant_ids.empty? && vm_tenant_ids.empty?
 
-    ["(vms.template = true AND vms.tenant_id IN (?)) OR (vms.template = false AND vms.tenant_id IN (?))",
-     template_tenant_ids, vm_tenant_ids]
+    tenant = user_or_group.current_tenant
+    tenant_vms       = "vms.template = false AND vms.tenant_id IN (?)"
+    public_templates = "vms.template = true AND vms.publicly_available = true"
+
+    if tenant.source_id
+      private_tenant_templates = "vms.template = true AND vms.tenant_id = (?) AND vms.publicly_available = false"
+      ["#{private_tenant_templates} OR #{public_templates} OR #{tenant_vms}", tenant.id, vm_tenant_ids]
```

Comment:
@alexander-demichev it means that
**if the tenant came from **cloud tenant mapping**** 
it will restrict templates only on from current tenant's user and simultaneously with private templates (`vms.publicly_available = false`).
and
all public templates will be displayed.

it means that it is change according origin RBAC rule - **is it intended ?**

Reply:
Yes, this is the behaviour we want, for tenant that comes from cloud tenant mapping.

Reply:
cool and what about templates and VMs which not related to cloud openstack ?
These RBAC (descendant and ancestor) templates and vms (vms.publicly available = nil) should behave in origin way.

we can have also infra providers in Miq and when user will have mapped tenant, what will happen with their VMs ?

and I would expected test in this way:
let's have:
```
current user with tenant T2 from cloud tenant mapping in tenant  tree - Root Tenant -> T1 -> T2 -> T3
```
```
VMs(from infra) in tenant T1, T2, T3
Templates(from infra) in tenant T1, T2, T3
```

expect:
user can see VMs from T2 and T3
and
user can see Templates from T2 and T1

I think you can use on existing test and update user to have mapped tenant.

thanks